### PR TITLE
WIP: health check

### PR DESCRIPTION
**Pull request by ozkriff**

### CODE REVIEW

File: exonum/src/api/public/system.rs [hunk 1]
```diff
@@ -12,15 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::time::SystemTime;
+use std::collections::HashMap;
+
+use crypto::PublicKey;
```
Comment (DarkEld3r) [inline exonum/src/api/public/system.rs]: Module `crypto` is part of Exonum, so it should be in the next section. :wink:

Comment (ozkriff) [reply exonum/src/api/public/system.rs]: Done

**Comment by vldm** [comment]
Can you provide some info about pull request:
(description, examples, and how we could test it will be enought).

File: exonum/src/blockchain/service.rs [hunk 2]
```diff
@@ -304,24 +310,32 @@ impl SharedNodeState {
...
+
     /// Update internal state, from `Node` State`
     pub fn update_node_state(&self, state: &State) {
-        for (p, c) in state.peers().iter() {
-            self.state
-                .write()
-                .expect("Expected write lock.")
-                .peers_info
-                .insert(c.addr(), *p);
+        for (&public_key, c) in state.peers().iter() {
```
Comment (alekseysidorov) [inline exonum/src/blockchain/service.rs]: Is not the `state.peers().values()` working here?

File: exonum/src/blockchain/mod.rs [hunk 3]
```diff
@@ -392,7 +392,9 @@ impl Blockchain {
         mount
     }
 
-    fn api_context(&self) -> ApiContext {
+    // TODO: I need the public key - how can i get it?
+    // is it ok to make this public?
+    pub(crate) fn api_context(&self) -> ApiContext {
```
Comment (alekseysidorov) [inline exonum/src/blockchain/mod.rs]: It seems that the method should be public and documented.
And later we may be add possibility to add the custom handlers to the `NodeBulder`.

**Comment by DarkEld3r** [comment]
Superseded by #405.

**Pull request closed without merging**